docs/articles/commands/command_attributes.md

```diff
@@ -0,0 +1,78 @@
+---
+uid: commands_command_attributes
+title: Command Attributes
+---
+
+## Built-In Attributes
+CommandsNext has a variety of built-in attributes to enhance your commands and provide some access control.
+The majority of these attributes can be applied to your command methods and command groups.
+
+- @DSharpPlus.CommandsNext.Attributes.AliasesAttribute
+- @DSharpPlus.CommandsNext.Attributes.CooldownAttribute
+- @DSharpPlus.CommandsNext.Attributes.DescriptionAttribute
+- @DSharpPlus.CommandsNext.Attributes.DontInjectAttribute
+- @DSharpPlus.CommandsNext.Attributes.HiddenAttribute
+- @DSharpPlus.CommandsNext.Attributes.ModuleLifespanAttribute
+- @DSharpPlus.CommandsNext.Attributes.PriorityAttribute
+- @DSharpPlus.CommandsNext.Attributes.RemainingTextAttribute
+- @DSharpPlus.CommandsNext.Attributes.RequireBotPermissionsAttribute
+- @DSharpPlus.CommandsNext.Attributes.RequireDirectMessageAttribute
+- @DSharpPlus.CommandsNext.Attributes.RequireGuildAttribute
+- @DSharpPlus.CommandsNext.Attributes.RequireNsfwAttribute
+- @DSharpPlus.CommandsNext.Attributes.RequireOwnerAttribute
+- @DSharpPlus.CommandsNext.Attributes.RequirePermissionsAttribute
+- @DSharpPlus.CommandsNext.Attributes.RequirePrefixesAttribute
+- @DSharpPlus.CommandsNext.Attributes.RequireRolesAttribute
+- @DSharpPlus.CommandsNext.Attributes.RequireUserPermissionsAttribute
+
+
+## Custom Attributes
+If the above attributes don't meet your needs, CommandsNext also gives you the option of writing your own!
+Simply create a new class which inherits from `CheckBaseAttribute` and implement the required method.
+
+Our example below will only allow a command to be ran during a specified year.
+```cs
+public class RequireYearAttribute : CheckBaseAttribute
+{
+    public int AllowedYear { get; private set; }
+
+    public RequireYearAttribute(int year)
+    {
+        AllowedYear = year;
+    }
+
+    public override async Task<bool> ExecuteCheckAsync(CommandContext ctx, bool help)
+    {
+        if (AllowedYear != DateTime.Now.Year)
+        {
+            await ctx.RespondAsync($"Only usable during year {AllowedYear}.");
```

Comment:
As a rule of thumb, a check should only ever return true or false, and should definitely not interact with the user. Perhaps a better idea would be to make a check that allows for using a command on even weekdays.

Filtering and handling check failures in-depth, including notifying the user *why* a check failed should be the job of an exception handler.